### DISCUSSION TITLE
tpot22 hive sensor --> server connection explanation or more info

### DISCUSSION THREAD

**Discussion by lcia-projects** [telekom-security/tpotce#1104] [Q&A] [answered]
i am needing to make a few changes to the hive sensor --> server communication.. i'm having some trouble wrapping my head around how thats working. I have looked through the discussion boards and issues and cant seem to find the answers i need.. so here i go.

- 1 - I'm assuming the sensor sends the server logs and data via logstash.. i used the wiki article https://github.com/telekom-security/tpotce/wiki/Reconfigure-logstash.conf but it doesnt seem to make sense to me because no where in it does it have my server IP address.. is there a different configuration i should look for? i also looked for a filebeat instance on the sensor and didnt see one. any suggestions would be appreciated. 

- 2 - on the server.. i guess its the same question.. i see logstash running.. can you send data from one logstash (on the sensor) to to be caught by logstash on the server? 

thank you very much, this is a great project.

**Comment by Bastibaard** [comment]
First of all I am not an expert, I'm just a consumer of this amazing product but I'll try to maybe give an answer to your question. If you take a look at https://github.com/telekom-security/tpotce#distributed-deployment you will read that the logstash logs are being sent from sensor to hive via an SSH-tunnel over a custom port 64305. If you look on your **hive** and type "netstat -tulpn" you will see that it's actively listening on that port to receive logs from your sensor(s).

![image](https://user-images.githubusercontent.com/92089291/169985588-bb13599a-15a3-45cd-9509-09f3ca30f613.png)

Look at the **/data/elk/logstash** directory **on your sensor** to find the SSH-keys and HIVE configuration it's using. There you will find the information you entered when deploying your sensor. If you want to change your HIVE or whatever, you can also just remove said directory and redeploy the sensor as per the documentation. 

I hope I could somewhat answer (part of) your question 😄

**Reply by lcia-projects** [reply]
thank you that helps some.. but still confused about the logstash.conf. 

in the logstash conf, the output section is
output {
  elasticsearch {
    hosts => ["elasticsearch:9200"]
    # With templates now being legacy we need to set the daily index with its template manually. Otherwise a new index might be created with differents settings configured thro>
    index => "logstash-%{+YYYY.MM.dd}"
    template => "/etc/logstash/tpot-template.json"
    template_overwrite => "true"
  }

.. but its sending it to a different server not the local elasticsearch:9200 docker container.. so there is a different configuration somewhere its reading.. and i cant seem to find that file.. or how its sending it to the hive server.

**Comment by lcia-projects** [comment, marked as answer]
i figured it out and have ti working.. i'm not sure its ideal.. but it works.

i created a completely seperate ES cluster not associated with the TPOT22 or Hive installations. I got it all up and running. 3 SSD nodes (hot ILM nodes) and 2 nodes with spinning drives(warm ILM nodes) for long term storage.

once that was all set, i stopped the tpot service. and followed the instructions on the WIKI  : https://github.com/telekom-security/tpotce/wiki/Reconfigure-logstash.conf

if you are doing an single node (not hive install) just follow those directions. with the hive server you need to switch the logstash.conf for the http_input.conf. 

thats really the only change. 

i still send my data to the hive elasticsearch. i just delete data after two weeks. its quick and easy. i just also send the data to my other es cluster for long term storange and analysis. 
this is the output section of the http_input.conf i'm using:

# Output section
output {
  ### short term storage (faster) and pew pew attack map on hive server
  elasticsearch {
    hosts => ["elasticsearch:9200"]
    # With templates now being legacy we need to set the daily index with its template manually. Otherwise a new index might be created with differents settings configured through Kibana.
    index => "logstash-%{+YYYY.MM.dd}"
    template => "/etc/logstash/tpot-template.json"
    template_overwrite => "true"
  }
  ### long term cluster storage
  elasticsearch {
    hosts => ["10.0.0.20:9200","10.0.0.21:9200","10.0.0.22:9200"]
    # With templates now being legacy we need to set the daily index with its template manually. Otherwise a new index might be created with differents settings configured through Kibana.
    index => "logstash-ds-tpot22-%{+YYYY.MM}"
    template => "/etc/logstash/tpot-template.json"
    template_overwrite => "true"
    action => "create" # this is needed for an es data-stream, its important
  }
}

i hope this helps

**Reply by lullu57** [reply]
Hi I'm trying to set up the connection from a base tpot iso build run on a vm, to a local elk stack. I tried changing the tpot's logstash output section as described above, however, I am not receiving any input on the local elk stack. Is there anything which I need to do on the elk stack to be able to receive the input from tpot?

**Comment by lcia-projects** [comment]
i set up an elk stack like i mentioned above. 
i used the index pattern from the internal elk stack that comes with tpot in my own cluster. 

depending on whether you are doing a hive install or a stand alone install will determine what .yml you need to modify. 
Check your elasticsearch logs and logstash logs to see what the issue is.

**Reply by lullu57** [reply]
Hi so I've been playing around trying to get this to work again with the configuration as described above but haven't managed to get the connection working. Could you kindly guide me as to where I can the tpot instance logstash logs are? The issue must be from there but I haven't located where they are yet. Thanks once again :)
